system/Database/BaseBuilder.php

```diff
@@ -2417,6 +2424,11 @@ public function updateBatch($set = null, $constraints = null, int $batchSize = 1
 
     /**
      * Generates a platform-specific batch update string from the supplied data
+     *
+     * @used-by batchExecute
+     *
+     * @param array $keys   QBKeys
+     * @param array $values QBSet
```

Comment:
Can these get any more specific? Let's probably are all strings, right?

Reply:
$values can be RawSql

Reply:
@sclubricants But it is `array` typed.
`protected function _updateBatch(string $table, array $keys, array $values): string`

Reply:
$QBKeys: `list<string>`
$QBSet: `array<string, string>|list<list<string|int>>`, but in this case `list<list<string|int>>` only.

Reply:
Updated.